include/openssl/x509_vfy.h

```diff
@@ -89,13 +89,21 @@ void X509_STORE_CTX_set_depth(X509_STORE_CTX *ctx, int depth);
 
 # define X509_L_FILE_LOAD        1
 # define X509_L_ADD_DIR          2
+# define X509_L_ADD_STORE          3
+# define X509_L_LOAD_STORE         4
 
 # define X509_LOOKUP_load_file(x,name,type) \
                 X509_LOOKUP_ctrl((x),X509_L_FILE_LOAD,(name),(long)(type),NULL)
 
 # define X509_LOOKUP_add_dir(x,name,type) \
                 X509_LOOKUP_ctrl((x),X509_L_ADD_DIR,(name),(long)(type),NULL)
 
+# define X509_LOOKUP_add_store(x,name) \
+                X509_LOOKUP_ctrl((x),X509_L_ADD_STORE,(name),0,NULL)
+
+# define X509_LOOKUP_load_store(x,name) \
+                X509_LOOKUP_ctrl((x),X509_L_LOAD_STORE,(name),0,NULL)
+
```

Comment:
We have "load" file and "add" directory. Store is similar to directory. This mean that "load" is extra.
By design goal of "load" is to fetch all cert/crl for a file, while "add" is to fetch on "request". Working with first one is like to "dump" whole external database (filtered by certain criteria) into memory , i.e. OpenSSL X.509 store cache.
I would like to suggest to keep only "ADD" and to remove "LOAD".

More over my test shows that "LOAD" does not work: control method calls cache_objects() with lookup type "NONE". This type is not supported and cache_objects() raises error with reason X509_R_WRONG_LOOKUP_TYPE .

Reply:
> Store is similar to directory.

Store is actually similar to *both* "file" (i.e. accessing a single object) and "directory" (accessing a pile of objects)

Reply:
We do need to keep "load" support.  This is called all over the place.

I think I did have support for "NONE" at some point...  I'll revisit that.